tests/ga/test_extension.py

```diff
@@ -376,185 +374,193 @@ def _set_up_update_test_and_update_gs(self, patch_command, *args):
         self._assert_ext_status(protocol.report_ext_status, "success", 0)
 
         # Next incarnation, update version
-        test_data.goal_state = test_data.goal_state.replace("<Incarnation>1<", "<Incarnation>2<")
-        test_data.ext_conf = test_data.ext_conf.replace('version="1.0.0"', 'version="1.0.1"')
-        test_data.manifest = test_data.manifest.replace('1.0.0', '1.0.1')
+        test_data.set_incarnation(2)
+        test_data.set_extensions_config_version("1.0.1")
+        test_data.set_manifest_version('1.0.1')
+        protocol.update_goal_state()
```

Comment:
now the call to update_goal_state is done in the main loop, instead of in the extension handler, so these tests need an explicit call to update_goal_state